src/DefaultBuilder/test/Microsoft.AspNetCore.TrimmingTests/X509Utilities.cs

```diff
@@ -0,0 +1,20 @@
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNetCore.Builder;
+
+#nullable enable
+
+public static class X509Utilities
+{
+    public static bool HasCertificateType =>
+        GetType("System.Security.Cryptography", "System.Security.Cryptography.X509Certificates.X509Certificate") is not null;
+
+    [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:RequiresUnreferencedCode",
+        Justification = "Returning null when the type is unreferenced is desirable")]
+    private static Type? GetType(string assemblyName, string typeName) {
+        var assembly = AppDomain.CurrentDomain.GetAssemblies().SingleOrDefault(a => a.GetName().Name == assemblyName);
+        return assembly?.GetType(typeName);
```

Comment:
Why not use `Type.GetType` here?

Reply:
It didn't work.  Happy to learn why and use a more appropriate API.

Reply:
What didn't work?

Reply:
AFAICT, it returned null for all `Microsoft.AspNetCore` types, regardless of whether or not they were preserved.  (It did consistently find `System.Threading.Tasks.Task` though.)

Reply:
The [runtime tests](https://github.com/dotnet/runtime/blob/e4743ec9474728f2b4e120fd70a3f729e9204c32/src/libraries/System.Text.Json/tests/System.Text.Json.Tests/TrimmingTests/IsReflectionEnabledByDefaultFalse.cs#L65) also appear to avoid `Type.GetType`, though that may be for some other reason.

Reply:
@eerhardt pointed out offline that it wasn't working because I wasn't passing the assembly name.  Fixed.